src/test/java/cucumber/GenerateJoinLinkStepdefs.java

```diff
@@ -0,0 +1,32 @@
+package cucumber;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.notNullValue;
+
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import io.restassured.response.Response;
+
+public class GenerateJoinLinkStepdefs {
+  private Response response;
+
+  @When("the generate join link endpoint is called")
+  public void theGenerateJoinLinkEndpointIsCalled() {
+    Long storedTeamId = TeamWorld.storedTeam.getId();
+    var requestBody = new GenerateLinkRequest(storedTeamId);
+
+    response =
+        given()
+            .contentType("application/json")
+            .body(requestBody)
+            .when()
+            .post("/generate-join-link");
+  }
+
+  @Then("a link is generated")
+  public void aLinkIsGenerated() {
+    response.then().assertThat().statusCode(201).body("token", notNullValue());
```

Comment:
Would it be redundant to include here a check for the teamId inside the token? I wonder if it is redundant because we already test something like that in the TokenService test

Reply:
well what we are actually returning is only the token not an actual object so we would have to check directly in the db which can be done for sure but yes might be overlapping with the unit tests that we already have

Reply:
👍